persistent/persistence.py

```diff
@@ -52,62 +54,69 @@
 # check in __getattribute__
 _SPECIAL_NAMES = set(SPECIAL_NAMES)
 
+_SLOTS = ('__jar', '__oid', '__serial', '__flags', '__size', '__ring',)
+_SPECIAL_NAMES.update([intern('_PersistentPy' + x) for x in _SLOTS])
+
+
 # Represent 8-byte OIDs as hex integer, just like
 # ZODB does.
 _OID_STRUCT = struct.Struct('>Q')
 _OID_UNPACK = _OID_STRUCT.unpack
 
-@implementer(IPersistent)
-class Persistent(object):
+# We don't apply ``use_c_impl`` as a decorator to a class named
+# Persistent because we need access to the PersistentPy name
+# in a few places in the implementation.
+@implementer(interfaces.IPersistent)
+class PersistentPy(object):
```

Comment:
I went back and forth on this. I originally stuck with the name `Persistent`, but it looked weird (and of course caused linter errors) to refer to `PersistentPy` in the body; and if you forgot, you'd get weird errors. I only went with `PersistentPy` when I started to write a comment explaining that the class was named `Persistent` but you had to refer to `PersistentPy` in the body.

The other alternative would be to simply not import the C extension into this module at all, just in `__init__.py`. But I like the symmetry of doing everything (nearly) the same across all the modules and treating the C extension as an implementation detail private to the module.